webrender/src/device/gfx/buffer.rs

```diff
@@ -2,15 +2,24 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+use arrayvec::ArrayVec;
 use hal;
 use hal::device::Device as BackendDevice;
+use rendy_descriptor::{DescriptorAllocator, DescriptorSet};
 use rendy_memory::{Block, Heaps, Kind, MappedRange, MemoryBlock, MemoryUsage, MemoryUsageValue, Write};
 
 use std::cell::Cell;
 use std::sync::Arc;
 use std::mem;
 
+use super::{MAX_FRAME_COUNT};
+use super::descriptor::{DescriptorData, DescriptorGroup};
+
 pub const DOWNLOAD_BUFFER_SIZE: usize = 10 << 20; // 10MB
+// Maximum number of bound projections per frame.
+// Projections are bound on a per render target basis, so those should fit in the 96 limit.
+// We need this limit to define a fixed size for our uniform buffers.
+pub const PROJECTION_PER_FRAME: usize = 96;
```

Comment:
where are we checking if we hit this limit?

Reply:
Added an assert for now https://github.com/szeged/webrender/pull/346/files#diff-5dbe2f57bb9273f9a363bcb71ec6834eR765